usecases/replica/finder.go

```diff
@@ -97,27 +103,86 @@ func (f *Finder) GetOne(ctx context.Context,
 			return findOneReply{host, x.Version, r, x.UpdateTime, true}, err
 		}
 	}
-	replyCh, state, err := c.Fetch2(ctx, l, op)
+	replyCh, state, err := c.Pull(ctx, l, op)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%s %q: %w", msgCLevel, l, err)
 	}
 	result := <-f.readOne(ctx, shard, id, replyCh, state)
-	return result.data, result.err
+	if err = result.err; err != nil {
+		err = fmt.Errorf("%s %q: %w", msgCLevel, l, err)
+	}
+	return result.data, err
+}
+
+// batchReply represents the data returned by sender
+// The returned data may result from a direct or digest read request
+type batchReply struct {
+	// Sender hostname of the sender
+	Sender string
+	// DigestRead is this reply from digest read?
+	DigestRead bool
```

Comment:
Maybe `IsDigest` or similar would work here. This way the meaning is obvious without needing to check the comment on the definition

Reply:
Agreed